packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx

```diff
@@ -0,0 +1,347 @@
+          </ActionList>
+        </DropdownOverlay>
+      </Dropdown>,
+    );
+
+    const selectInput = getByRole('combobox', { name: 'Fruits' });
+
+    expect(selectInput).toBeInTheDocument();
+    expect(container.querySelector('[role=listbox]')).not.toBeVisible();
+
+    await user.click(selectInput);
+    expect(getByRole('listbox', { name: 'Fruits' })).toBeVisible();
+
+    await user.keyboard('{Escape}');
+    expect(getByRole('listbox', { name: 'Fruits' })).not.toBeVisible();
+  });
+
+  it('should move focus between items with arrow key', async () => {
+    const user = userEvent.setup();
+    const { container, getByRole } = renderWithTheme(
+      <Dropdown>
+        <SelectInput label="Fruits" />
+        <DropdownOverlay>
+          <ActionList>
+            <ActionListItem title="Apple" value="apple" />
+            <ActionListItem title="Mango" value="mango" />
+          </ActionList>
+        </DropdownOverlay>
+      </Dropdown>,
+    );
+
+    const selectInput = getByRole('combobox', { name: 'Fruits' });
+
+    expect(selectInput).toBeInTheDocument();
+    expect(selectInput.textContent).toBe('Select Option');
+    expect(container.querySelector('[role=listbox]')).not.toBeVisible();
+
+    const getActiveDescendant = (): string | null | undefined => {
+      const activeDescendantId = selectInput.getAttribute('aria-activedescendant');
+      const activeDescendantElement = container.querySelector(`#${activeDescendantId}`);
+      return activeDescendantElement?.textContent;
+    };
+
+    // Dropdown open
+    selectInput.focus();
+    await user.keyboard('{ArrowDown}');
+    expect(getByRole('listbox', { name: 'Fruits' })).toBeVisible();
+
+    // Move to first item
+    await user.keyboard('{ArrowDown}');
+    expect(getActiveDescendant()).toBe('Apple');
+
+    // Move to second item
+    await user.keyboard('{ArrowDown}');
+    expect(getActiveDescendant()).toBe('Mango');
+
+    // Select option
+    await user.keyboard('[Space]');
+    expect(selectInput.textContent).toBe('Mango');
+  });
+
+  it('should move focus between items with arrow key in multiselect', async () => {
+    const user = userEvent.setup();
+    const { container, getByRole } = renderWithTheme(
+      <Dropdown selectionType="multiple">
+        <SelectInput label="Fruits" />
+        <DropdownOverlay>
+          <ActionList>
+            <ActionListItem title="Apple" value="apple" />
+            <ActionListItem title="Mango" value="mango" />
+            <ActionListItem title="Orange" value="orange" />
+          </ActionList>
+        </DropdownOverlay>
+      </Dropdown>,
+    );
+
+    const selectInput = getByRole('combobox', { name: 'Fruits' });
+
+    expect(selectInput).toBeInTheDocument();
+    expect(selectInput.textContent).toBe('Select Option');
+    expect(container.querySelector('[role=listbox]')).not.toBeVisible();
+
+    const getActiveDescendant = (): string | null | undefined => {
```

Comment:
Move this outside maybe? We are redeclaring again.

Reply:
done